[BUG] Some slabs don't cover tracks

**Describe the bug**
Placing slabs on track create error. Some slabs don't render at all.

**To Reproduce**
Steps to reproduce the behavior:
1. Place tracks
2. Take slab
3. Right click on track with slab
4. See (or not)

**Expected behavior**
Render the cover without having any error.

**Screenshots**
```
[19:03:45] [Flywheel Task Executor #4/ERROR] [flywheel/]: Error running task
net.minecraft.ReportedException: Tesselating block model
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.block.ModelBlockRenderer.tesselateBlock(ModelBlockRenderer.java:91) ~[client-1.21.1-20240808.144430-srg.jar%23300!/:?] {re:mixin,pl:accesstransformer:B,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:copycats-common.mixins.json:foundation.copycat.multistate.ModelBlockRendererMixin from mod copycats,pl:mixin:APP:sodium-common.mixins.json:features.render.frapi.ModelBlockRendererMixin from mod sodium,pl:mixin:APP:sodium-common.mixins.json:features.textures.animations.tracking.ModelBlockRendererMixin from mod sodium,pl:mixin:APP:sodium-neoforge.mixins.json:features.render.model.block.ModelBlockRendererMixin from mod sodium,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.lib.model.baked.BakedModelBufferer.bufferModel(BakedModelBufferer.java:59) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.lib.model.baked.ModelBuilderImpl.buildBakedModelBuilder(ModelBuilderImpl.java:20) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.impl.FlwLibXplatImpl.buildBakedModelBuilder(FlwLibXplatImpl.java:24) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.lib.model.baked.BakedModelBuilder.build(BakedModelBuilder.java:64) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/create@6.0.8/com.simibubi.create.foundation.render.SpecialModels.lambda$static$1(SpecialModels.java:25) ~[create-1.21.1-6.0.8.jar%23329!/:6.0.8] {re:classloading,re:mixin}
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708) ~[?:?] {}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.lib.util.RendererReloadCache.get(RendererReloadCache.java:26) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/create@6.0.8/com.simibubi.create.foundation.render.SpecialModels.smoothLit(SpecialModels.java:32) ~[create-1.21.1-6.0.8.jar%23329!/:6.0.8] {re:classloading,re:mixin}
	at TRANSFORMER/railways@0.1.0+neoforge-mc1.21.1/com.railwayteam.railways.content.custom_tracks.casing.CasingRenderUtils.makeCasingInstance(CasingRenderUtils.java:188) ~[railways-0.1.0+neoforge-mc1.21.1.jar%23406!/:0.1.0+neoforge-mc1.21.1] {re:mixin,re:classloading}
	at TRANSFORMER/create@6.0.8/com.simibubi.create.content.trains.track.TrackVisual.railways$makeCasingData(TrackVisual.java:1185) ~[create-1.21.1-6.0.8.jar%23329!/:6.0.8] {re:mixin,re:classloading,pl:mixin:APP:railways-common.mixins.json:client.AccessorTrackVisual from mod railways,pl:mixin:APP:railways-common.mixins.json:client.MixinTrackVisual from mod railways,pl:mixin:A}
	at TRANSFORMER/create@6.0.8/com.simibubi.create.content.trains.track.TrackVisual.handler$dpl000$railways$onCtor(TrackVisual.java:1116) ~[create-1.21.1-6.0.8.jar%23329!/:6.0.8] {re:mixin,re:classloading,pl:mixin:APP:railways-common.mixins.json:client.AccessorTrackVisual from mod railways,pl:mixin:APP:railways-common.mixins.json:client.MixinTrackVisual from mod railways,pl:mixin:A}
	at TRANSFORMER/create@6.0.8/com.simibubi.create.content.trains.track.TrackVisual.<init>(TrackVisual.java:52) ~[create-1.21.1-6.0.8.jar%23329!/:6.0.8] {re:mixin,re:classloading,pl:mixin:APP:railways-common.mixins.json:client.AccessorTrackVisual from mod railways,pl:mixin:APP:railways-common.mixins.json:client.MixinTrackVisual from mod railways,pl:mixin:A}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.lib.visualization.SimpleBlockEntityVisualizer.createVisual(SimpleBlockEntityVisualizer.java:27) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.impl.visualization.storage.BlockEntityStorage.createRaw(BlockEntityStorage.java:55) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.impl.visualization.storage.BlockEntityStorage.createRaw(BlockEntityStorage.java:15) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.impl.visualization.storage.Storage.add(Storage.java:72) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.impl.visualization.VisualManagerImpl.processQueue(VisualManagerImpl.java:63) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.impl.visualization.VisualManagerImpl.lambda$framePlan$0(VisualManagerImpl.java:71) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.lib.task.Distribute.tasks(Distribute.java:41) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.lib.task.SimplePlan.lambda$execute$0(SimplePlan.java:33) ~[flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.impl.task.ParallelTaskExecutor$WorkerThread.processTask(ParallelTaskExecutor.java:215) [flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
	at TRANSFORMER/flywheel@1.0.5/dev.engine_room.flywheel.impl.task.ParallelTaskExecutor$WorkerThread.run(ParallelTaskExecutor.java:205) [flywheel-neoforge-1.21.1-1.0.5.jar%23580!/:?] {re:classloading}
Caused by: java.lang.IllegalArgumentException: Cannot get property EnumProperty{name=type, clazz=class net.minecraft.world.level.block.state.properties.SlabType, values=[top, bottom, double]} as it does not exist in Block{minecraft:air}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.level.block.state.StateHolder.getValue(StateHolder.java:97) ~[client-1.21.1-20240808.144430-srg.jar%23300!/:?] {re:mixin,pl:connector_pre_launch:A,re:computing_frames,pl:connector_pre_launch:A,re:classloading,pl:mixin:APP:ferritecore.fastmap.mixin.json:FastMapStateHolderMixin from mod ferritecore,pl:mixin:APP:create.mixins.json:accessor.StateHolderAccessor from mod create,pl:mixin:A,pl:connector_pre_launch:A}
	at TRANSFORMER/ferritecore@7.0.2/malte0811.ferritecore.impl.KeyValueConditionImpl.lambda$getBlockStatePredicate$1(KeyValueConditionImpl.java:86) ~[ferritecore-7.0.2-neoforge.jar%23375!/:?] {re:mixin,re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.resources.model.MultiPartBakedModel.getQuads(MultiPartBakedModel.java:1074) ~[client-1.21.1-20240808.144430-srg.jar%23300!/:?] {re:mixin,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:fabric-renderer-api-v1.mixins.json:client.MultipartBakedModelMixin from mod fabric_renderer_api_v1,pl:mixin:APP:sodium-neoforge.mixins.json:features.model.MultiPartBakedModelMixin from mod sodium,pl:mixin:APP:ferritecore.dedupmultipart.mixin.json:MixinMultipartModel from mod ferritecore,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/neoforge@21.1.213/net.neoforged.neoforge.client.model.IDynamicBakedModel.getQuads(IDynamicBakedModel.java:25) ~[neoforge-21.1.213-universal.jar%23301!/:?] {re:mixin,re:computing_frames,re:classloading}
	at TRANSFORMER/fabric_renderer_api_v1@3.4.0+acb05a3919/net.fabricmc.fabric.api.renderer.v1.model.ForwardingBakedModel.getQuads(ForwardingBakedModel.java:67) ~[fabric-renderer-api-v1-3.4.0+acb05a3919.jar%23561!/:3.4.0+acb05a3919] {re:classloading}
	at TRANSFORMER/railways@0.1.0+neoforge-mc1.21.1/com.railwayteam.railways.content.custom_tracks.casing.SpriteCopyingBakedModel.getQuads(SpriteCopyingBakedModel.java:50) ~[railways-0.1.0+neoforge-mc1.21.1.jar%23406!/:0.1.0+neoforge-mc1.21.1] {re:classloading}
	at TRANSFORMER/neoforge@21.1.213/net.neoforged.neoforge.client.extensions.IBakedModelExtension.getQuads(IBakedModelExtension.java:41) ~[neoforge-21.1.213-universal.jar%23301!/:?] {re:classloading,re:mixin}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.block.ModelBlockRenderer.tesselateWithAO(ModelBlockRenderer.java:131) ~[client-1.21.1-20240808.144430-srg.jar%23300!/:?] {re:mixin,pl:accesstransformer:B,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:copycats-common.mixins.json:foundation.copycat.multistate.ModelBlockRendererMixin from mod copycats,pl:mixin:APP:sodium-common.mixins.json:features.render.frapi.ModelBlockRendererMixin from mod sodium,pl:mixin:APP:sodium-common.mixins.json:features.textures.animations.tracking.ModelBlockRendererMixin from mod sodium,pl:mixin:APP:sodium-neoforge.mixins.json:features.render.model.block.ModelBlockRendererMixin from mod sodium,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
	at TRANSFORMER/neoforge@21.1.213/net.neoforged.neoforge.client.model.lighting.LightPipelineAwareModelBlockRenderer.tesselateWithAO(LightPipelineAwareModelBlockRenderer.java:56) ~[neoforge-21.1.213-universal.jar%23301!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.client.renderer.block.ModelBlockRenderer.tesselateBlock(ModelBlockRenderer.java:82) ~[client-1.21.1-20240808.144430-srg.jar%23300!/:?] {re:mixin,pl:accesstransformer:B,pl:connector_pre_launch:A,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:copycats-common.mixins.json:foundation.copycat.multistate.ModelBlockRendererMixin from mod copycats,pl:mixin:APP:sodium-common.mixins.json:features.render.frapi.ModelBlockRendererMixin from mod sodium,pl:mixin:APP:sodium-common.mixins.json:features.textures.animations.tracking.ModelBlockRendererMixin from mod sodium,pl:mixin:APP:sodium-neoforge.mixins.json:features.render.model.block.ModelBlockRendererMixin from mod sodium,pl:mixin:A,pl:connector_pre_launch:A,pl:runtimedistcleaner:A}
	... 22 more
```

--- a/src/main/java/com/railwayteam/railways/content/custom_tracks/casing/CasingRenderUtils.java
+++ b/src/main/java/com/railwayteam/railways/content/custom_tracks/casing/CasingRenderUtils.java
@@ -68,8 +68,9 @@ public abstract class CasingRenderUtils {
     public static PartialModel reTexture(PartialModel model, SlabBlock block) {
         Pair<PartialModel, SlabBlock> key = Pair.of(model, block);
         if (!reTexturedModels.containsKey(key)) {
-            BakedModel slabModel = Minecraft.getInstance().getModelManager().getBlockModelShaper().getBlockModel(block.defaultBlockState());
-            BakedModel texturedCasing = new SpriteCopyingBakedModel(model.get(), slabModel);
+            BlockState slabState = block.defaultBlockState();
+            BakedModel slabModel = Minecraft.getInstance().getModelManager().getBlockModelShaper().getBlockModel(slabState);
+            BakedModel texturedCasing = new SpriteCopyingBakedModel(model.get(), slabModel, slabState);
             PartialModel texturedPartial = RuntimeFakePartialModel.make(Railways.asResource("runtime_casing"), texturedCasing);
             reTexturedModels.put(key, texturedPartial);
             return texturedPartial;

--- a/src/main/java/com/railwayteam/railways/content/custom_tracks/casing/SpriteCopyingBakedModel.java
+++ b/src/main/java/com/railwayteam/railways/content/custom_tracks/casing/SpriteCopyingBakedModel.java
@@ -36,10 +36,12 @@ public class SpriteCopyingBakedModel implements BakedModel {
 
   protected final BakedModel baseModel;
   protected final BakedModel spriteSourceModel;
+  protected final BlockState spriteSourceState;
 
-  public SpriteCopyingBakedModel(BakedModel baseModel, BakedModel spriteSourceModel) {
+  public SpriteCopyingBakedModel(BakedModel baseModel, BakedModel spriteSourceModel, BlockState spriteSourceState) {
     this.baseModel = baseModel;
     this.spriteSourceModel = spriteSourceModel;
+    this.spriteSourceState = spriteSourceState;
   }
 
   @Override
@@ -47,13 +49,12 @@ public class SpriteCopyingBakedModel implements BakedModel {
     ArrayList<BakedQuad> quads = new ArrayList<>();
     TextureAtlasSprite overrideSprite = spriteSourceModel.getParticleIcon();
     BakedQuad overrideQuad = null;
-    List<BakedQuad> sourceQuads = spriteSourceModel.getQuads(pState, pSide, pRand);
+    List<BakedQuad> sourceQuads = spriteSourceModel.getQuads(spriteSourceState, pSide, pRand);
     if (!sourceQuads.isEmpty()) {
       overrideSprite = sourceQuads.get(0).getSprite();
       overrideQuad = sourceQuads.get(0);
-      //Railways.LOGGER.warn("Overridesprite: "+ overrideSprite.toString());
     } else if (pSide != null) {
-      List<BakedQuad> nullQuads = spriteSourceModel.getQuads(pState, null, pRand);
+      List<BakedQuad> nullQuads = spriteSourceModel.getQuads(spriteSourceState, null, pRand);
       if (!nullQuads.isEmpty()) {
         overrideSprite = nullQuads.get(0).getSprite();
         overrideQuad = nullQuads.get(0);

